# Maternal_Effect_ID


The goal of this project is to identify genes that are strictly maternally contributed to the embryo to use as a means of insect control.  The genes must only be expressed by the mother to initiate development in the next generationâ€™s embryo.  To identify candidate loci, we combined data from a set of NimbleGen tiling array databases, with a computational gene model prediction database.  The NimbleGen tiling arrays measure the transcript abundance of every 400 bp segment of the genome in the sample being tested.  We interfaced these global expression databases with gene prediction models to parse out transcript abundance of every possible locus in the genome (14394).  This was done at 10 differing life stages, 6 hour embryos, Early Adult Females, Early Adult Males, Early Last Larval Instars, Early Pupal Females, Early Pupal Males, 
Late Last Larval Instars, Mid Last Larval Instars, Mid Pupal Females, Mid Pupal Males. The loci of interest are the the ones that maximize the difference between 6 hour Embryos and all other life stages.  Each gene has 9 stage comparisons, and each stage comparison has a number of observations equal to the length of the gene/400. Maternal Effect Strictness was modeled using a Linear mixed models with gene ID as a random effect.  The best linear unbiased predictors (BLUP estimates) were extracted from the model for each gene.  The genes with the highest BLUP values were confirmed to have no non-embryonic fitness effects using an RNAi knockdown database and passed on to a collaborator to begin population genetic engineering experiments. 
